manifests/init.pp

```diff
@@ -1,7 +1,7 @@
 # Manage an ISC DHCP server
 class dhcp (
   Array[String] $dnsdomain = $dhcp::params::dnsdomain,
-  Array[String] $nameservers = ['8.8.8.8', '8.8.4.4'],
+  Optional[Array[String]] $nameservers = ['8.8.8.8', '8.8.4.4'],
```

Comment:
IMHO we should default to an empty array instead and skip the option when it's empty. You can't go back to undef if there's a default.

Reply:
Indeed. I'm more than happy to remove the default. To be pedantic, do you want to allow `undef` or restrict it to `[]`.

Reply:
IMHO `Array[String]` or even `Array[Stdlib::Host]`. For this module an empty array is equivalent to undef and and code is easier to deal with when you can assume an array.

Reply:
That works for me.  I have revised the PR but there is some awkwardness in that if either `dnsupdatekey` or `ddns_updates` is set, `dhcpd.conf.erb` expects that `dnsupdateserver_real` will have a real value.